src/lib/translators/mongo.ts

```diff
@@ -173,16 +172,14 @@ function transformPercentage(step: PercentageStep): Array<MongoStep> {
 
 /** transform an 'replace' step into corresponding mongo steps */
 function transformReplace(step: ReplaceStep): MongoStep {
+  const branches: Array<MongoStep> = [];
+  for (const k in step.to_replace) {
```

Comment:
I think this could be written as:

```typescript
  const branches: Array<MongoStep> = [];
  for (const [oldval, newval] of Object.entries(step.to_replace)) {
    branches.push({ case: { $eq: [`$${step.search_column}`, oldval] }, then: newval });
  }
```

or directly with a `map`:

```typescript
    const branches: Array<MongoStep> = Object.entries(step.to_replace).map((oldval, newval) => (
        { case: { $eq: [`$${step.search_column}`, oldval] }, then: newval }
    );
```

Reply:
Nice ! Thank you for the suggestion :)

Reply:
Nice ! Thank you for the suggestion :)